zkevm-circuits/src/pi_circuit.rs

```diff
@@ -150,6 +160,23 @@ impl PublicData {
         num_all_txs_in_blocks
     }
 
+    fn get_num_all_l1_block_hashes(&self) -> usize {
+        let mut num_all_l1_block_hashes_in_blocks = 0;
+        for block_ctx in self.block_ctxs.ctxs.values() {
+            if let Some(l1_block_hashes) = &block_ctx.l1_block_hashes {
+                num_all_l1_block_hashes_in_blocks += l1_block_hashes.len();
+            }
+        }
+
+        num_all_l1_block_hashes_in_blocks
```

Comment:
can't we just do `last_applied_l1_block - prev_last_applied_l1_block` to get the same number?

Reply:
we want the exact count of l1 block hashes so we can constraint it against `last_applied_l1_block - prev_last_applied_l1_block`